cmk/base/plugins/agent_based/checkpoint_connections.py

```diff
@@ -53,14 +64,33 @@ def check_checkpoint_connections(
     params,
     section: Section,
 ) -> CheckResult:
+
+    if section.maximum > 0 and "relative_levels" in params:
+        warn_pct, crit_pct = params["relative_levels"]
+        warn = section.maximum * warn_pct / 100
+        crit = section.maximum * crit_pct / 100
+        levels_upper = (warn, crit)
+    else:
+        # use absolute levels if no relative levels provided or no maximum set on CP
+        levels_upper = params["levels"]
     yield from check_levels(
         value=section.current,
-        levels_upper=params["levels"],
+        levels_upper=levels_upper,
         metric_name="connections",
         label="Current connections",
         render_func=str,
     )
 
+    yield Result(state=State.OK, summary="Peak: {0.peak}".format(section))
```

Comment:
This line and the line below can be combined into a single call to `check_levels`

Reply:
oh ok, I'll check the source code of the various check_levels definitions. Would I include the peak in the check_levels call above or add another 'yield from check_levels' for the "peak" value where no levels are defined? (https://docs.checkmk.com/master/en/devel_check_plugins.html mentions check_levels but doesn't yet have any documentation on it :))